internal/provider/auth_token_file.go

```diff
@@ -41,9 +41,8 @@ func GetTokenFileSchemaResource(authField string) *schema.Resource {
 	return mustAddLoginSchema(&schema.Resource{
 		Schema: map[string]*schema.Schema{
 			consts.FieldFilename: {
-				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc(consts.EnvVarTokenFilename, nil),
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
Correction to the above which is not entirely true. For fields that we previously set from a DefaultFunc and env vars we cannot set them as Required anymore. This is because we are handing setting them from env vars after the Provider has verified all required fields are set. So, for example, if we set this field to Required we would get a `Error: Missing required argument` error.